### DISCUSSION TITLE
UR in catch! is just showing UR 0

### DISCUSSION THREAD

**Discussion by greenmg2003** [ppy/osu#16596] [Q&A] [answered]
I don't know why, but I feel strange. In osu! stable, there are Unstable rate for all 4 rulesets, but in osu! lazer, catch! ruleset doesn't count UR (Even I selected "Show UR" in lazer! skin editor, it still says UR 0, even when I miss the drop/fruit/droplet). I wonder why, or this is a bug? And should this thing be updated?
Thanks for reading and I hope I will get answer soon!

**Comment by peppy** [comment, marked as answer]
What are you expecting to see? osu!catch has no timing accuracy.

**Reply by bdach** [reply]
To expand on this, stable's "unstable rate" for everything but catch is 10 times the standard deviation of the timing error for each object. But because catch is unique in the fact that its objects are not timed (as in, you either catch it or you don't), "unstable rate" in catch on stable uses the positional offset of the caught fruit for standard deviation (as in 0 UR would mean that every single object was caught at the exact same position of the catcher plate). This wasn't ported across (yet?) because it felt weird and was pretty much useless.

**Reply by greenmg2003** [reply]
Ah, I see why there are no UR in the catch! ruleset currently. Thank you!

**Comment by greenmg2003** [comment]
And in another word, what I'm expecting (if possible) is that, imagine that the catcher is just the rectangle box with all pixels of it are colored white, and then you color the border blue/green/etc... . If you suddenly move the catcher right next to the fruit, but it still doesn't hit the ground, and you hit the fruit in the left/right border of your catcher, then instead of 300, you earn 100. Same to drops, instead of 100, you get 50. And for droplets, it's okay to get 50 as long as it didn't hit the ground. And yeah, "late" catch like this will make the UR there.
That's what I mean. Thanks for reading!

**Reply by greenmg2003** [reply]
Seems like my idea is too crazy ._."

**Reply by greenmg2003** [reply]
@peppy (Sorry for bothering you again) Should this be okay enough for my idea of UR system? Or we just keep the system that hitting fruits on the top border will get a full score of it, and miss if the player absolutely leaves fruits dropped on the ground, or hits the fruits/drops/droplets on the left/right border?

**Reply by peppy** [reply]
I'm not against changing osu!catch judgement mechanics, but that's a deeper discussion.

**Reply by greenmg2003** [reply]
Alright, thanks! I'll hear the other news of the judgement of catch! rulesets from you and other people too soon!
